Doc/library/importlib.rst

```diff
@@ -420,30 +420,53 @@ ABC hierarchy::
         reloaded):
 
         - :attr:`__name__`
-            The name of the module.
+            The module's fully-qualified name.
+            It is ``'__main__'`` for non-package modules run from the file
+            system, non-package modules run from standard input and non-package
+            modules run from the module namespace.
 
         - :attr:`__file__`
-            The path to where the module data is stored (not set for built-in
-            modules).
+            The place from which the module's data was imported.  Its value and
+            meaning is up to the finder.
+            It is the relative path for modules run from the file system, the
+            absolute path for imported modules and modules run from the module
+            namespace and ``'<stdin>'`` for non-package modules run from
+            standard input.
+            It is not set for imported namespace packages, imported built-in
+            modules and imported frozen modules.
 
         - :attr:`__cached__`
-            The path to where a compiled version of the module is/should be
-            stored (not set when the attribute would be inappropriate).
+            The location of the module's cached data.
+            It is the relative path for packages run from the file system and
+            the absolute path for imported modules and modules run from the
+            module namespace.
+            It is not set for non-package modules run from the file system,
+            non-package modules run from standard input, imported namespace
+            packages, imported built-in modules and imported frozen modules.
 
         - :attr:`__path__`
-            A list of strings specifying the search path within a
```

Comment:
Alright.

Reply:
I am reusing the same wording that you used for `ModuleSpec` attributes below.

Reply:
I'm mostly fine with the change.  (I especially like the "indicator that a module is a package" part.)  There's just one point of accuracy that I think is worth keeping: it is a list of search locations rather than just file paths (though almost always only that).

So I'd say change the first line to be something like:

```
The list of locations where the package's submodules will be found.
```

Reply:
Thanks @ericsnowcurrently! I agree, I have just applied your suggested change. By the way, should we also use the more general term ‘location’ instead of ‘path’ for the `__cached__` attribute or in that case it can never be anything else than a path?